#!/bin/bash
#$ -V
#$ -cwd
#$ -j y
#$ -o kmer.$JOB_ID.log

################################ Description ####################################################

# This bash script performs k-mer analysis on Illumina sequencing reads of blowflies. 
#It uses KMC to count the k-mers and GenomeScope 2 to generate a report and informative plots.

#Inputs:

# Illumina sequencing (untrimmed) reads in .fastq.gz format located in 
# /data/ross/sequencing/raw/blowflies/picard_lab_illumina directory.

# Outputs:

# K-mer counts and histograms generated by KMC in the ./kmc directory.
# GenomeScope 2 report and informative plots in the ./genomescope directory.
# A log file in /data/ross/flies/analyses/blowflies/03_kmer/01_illumina_reads/logs directory.

# Author: Elpida Skarlou
# Date: 2023-03-27

#################################################################################################

set -e 

# PATHS

SCRATCH=/scratch/$USER/$JOB_ID/Kmer 
PIC_ILLUMINA=/data/ross/sequencing/raw/blowflies/picard_lab_illumina
KMER=/data/ross/flies/analyses/blowflies/03_kmer/01_illumina_reads/

# make $SCRATCH if it doesn't exist
mkdir -p $SCRATCH 

cd $SCRATCH

# transfer $TRIMMED/*.trimmed.fastq.gz to $PWD
rsync -av $PIC_ILLUMINA/*.fastq.gz .


### KMC ### 
# count k-mers in FASTQ files

mkdir kmc

# conda activate

# initialize conda
eval "$(conda shell.bash hook)"

# activate your conda environment
conda activate KMC


for file in $(ls *1_001.fastq.gz)
do
  base=$(basename $file "1_001.fastq.gz")
  cat ${base}1_001.fastq.gz ${base}2_001.fastq.gz > ${base}_files.fastq.gz
  kmc -k21 -t10 -m64 -ci1 -cs10000 -fq ${base}_files.fastq.gz ${base}_kmer_counts ./kmc   
  kmc_tools transform ${base}_kmer_counts histogram ${base}_kmer_k21.hist -cx100000
done


### Genome Scope 2 ### 
# produces a report and several informative plots describing the genome properties

# copy genomescope.R to SCRATCH (.)
cp /ceph/users/eskarlou/genomescope2.0/genomescope.R .

conda activate for_genomescope

mkdir genomescope


for file in $(ls *.hist)
do
  /ceph/users/eskarlou/miniconda3/envs/for_genomescope/bin/Rscript/genomescope.R -i ./kmc/$file -o ./genomescope -k 21
done


# syncing to final destinations #

# log file
mv $KMER/scripts/kmer.$JOB_ID.log $KMER/logs

# kmc
rsync -av ./kmc $KMER/outputs/

# genomescope
rsync -av ./genomescope $KMER/outputs/

rm -rf *
rm -rf /scratch/$USER/$JOB_ID